server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -168,10 +171,33 @@ public static ServerAddress startServer(MetricsSystem metricsSystem, ServerConte
           addresses);
     } catch (TTransportException e) {
       if (portSearch) {
+        // Build a list of reserved ports - as identified by properties of type PropertyType.PORT
+        ArrayList<Integer> restrictedPorts = new ArrayList<Integer>();
+        for (Property p : Property.values()) {
+          if (p.getType().equals(PropertyType.PORT)) {
+            for (int element : config.getPort(p)) {
+              restrictedPorts.add(element);
+              if (log.isDebugEnabled()) {
+                log.debug("Adding port {} to list of restricted ports", element);
```

Comment:
I don't think we need this debug statement here, but if there's an easy way to unit test the code with a mock configuration, that might be nice. Might need to move some of this port search logic into a separate method to make that easier.

Reply:
Done. I added what I think is a pretty comprehensive unit test. Please take a look when possible.